components/greaselion/browser/greaselion_extension_converter.cc

```diff
@@ -0,0 +1,133 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/greaselion/browser/greaselion_extension_converter.h"
+
+#include <stddef.h>
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/base64.h"
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/files/scoped_temp_dir.h"
+#include "base/json/json_file_value_serializer.h"
+#include "base/strings/string_util.h"
+#include "base/strings/utf_string_conversions.h"
+#include "base/values.h"
+#include "brave/common/network_constants.h"
+#include "brave/components/greaselion/browser/greaselion_download_service.h"
+#include "chrome/common/chrome_paths.h"
+#include "crypto/sha2.h"
+#include "extensions/common/constants.h"
+#include "extensions/common/extension.h"
+#include "extensions/common/file_util.h"
+#include "extensions/common/manifest_constants.h"
+#include "url/gurl.h"
+
+using extensions::Extension;
+using extensions::Manifest;
+
+namespace greaselion {
+
+scoped_refptr<Extension> ConvertGreaselionRuleToExtensionOnTaskRunner(
```

Comment:
does this need to be public? Can it just be in an anonymous namespace in GreaselionServiceImpl?